src/main/java/senla/com/socialnetwork/security/jwt/JwtTokenUtils.java

```diff
@@ -0,0 +1,66 @@
+package senla.com.socialnetwork.security.jwt;
+
+import io.jsonwebtoken.*;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import java.time.Duration;
+import java.util.Date;
+
+@Component
+@Log4j2
+public class JwtTokenUtils {
+
+    @Value("${app.jwt.secret}")
+    private String secret;
+
+    @Value("${app.jwt.lifetime}")
+    private Duration lifetimeAccessToken;
+
+
+    public String generateToken(UserDetails userDetails) {
+        Date issuedDate = new Date();
+        Date expiredDate = new Date(issuedDate.getTime() + lifetimeAccessToken.toMillis());
+        return Jwts.builder()
+                .setSubject(userDetails.getUsername())
+                .setIssuedAt(issuedDate)
+                .setExpiration(expiredDate)
+                .signWith(SignatureAlgorithm.HS256, secret)
+                .compact();
+    }
+
+    public String getUsername(String token) {
+        return getAllClaimsFromToken(token).getSubject();
+    }
+
+    private Claims getAllClaimsFromToken(String token) {
+        return Jwts.parser()
+                .setSigningKey(secret)
+                .parseClaimsJws(token)
+                .getBody();
+    }
+
+    public boolean validate(String token) {
+        try {
+            Jwts.parser()
+                    .setSigningKey(secret)
+                    .parseClaimsJws(token);
+            return true;
+        } catch (ExpiredJwtException expEx) {
```

Comment:
better use a switch or strategy pattern in such cases. It is preferable to use strategy. If you need explanation, I'll provide later